### ISSUE TITLE
MIP sensitivity labels don't appear to save

### ISSUE THREAD

**Issue by PD-LUHC** [JanMarvin/openxlsx2#1144]
Having followed the thread of issues starting at https://github.com/ycphs/openxlsx/issues/461 and ending at https://github.com/JanMarvin/openxlsx2/pull/983 I have had limited success in getting MIP labels to successfully apply to files saved using `wb_save`. Possibly I'm being dim, but I don't see why the simplified, example below shouldn't work:

```R
xlsx_path <- "~/simple_test2.xlsx" # desired output path
wb_w_mips <- wb_load("~/mip_template.xlsx") # existing xlsx with correct mip label
mip_xml <- wb_get_mips(wb_w_mips) # also tested this by manually extracting LabelInfo.xml and pasting string in here - no joy.

wb <- wb_workbook() |>
    wb_add_worksheet() |> 
    wb_add_mips(xml = mip_xml) |>
    wb_save(file = xlsx_path, overwrite = TRUE) # this saves a file with no mip label set

if (interactive()) wb$open() # this opens a temp file with mip label set
```

Interestingly saving the file does not appear to apply the label. But using `open()` launches a temp file with the label applied. Is it just me or is that unintuitive? What is it about saving a fresh file that prevents the mip label being applied?
![image](https://github.com/user-attachments/assets/bec15b98-29af-47b6-b455-55fc18a14fc3)
![image](https://github.com/user-attachments/assets/2108a21a-05ba-4a99-9069-a67f895d0a8f)

This is running on openxlsx2 1.9

**Comment by JanMarvin** [comment]
Hi @PD-LUHC , your code looks valid and I’m not sure what happens. The difference between `wb$open()` and `wb$save()` should be that the first writes a file to tempdir and opens it and the latter simply saves the file.

As stated before, I have no way to test this. I can look into it sometime next week, but unless something strikes my eye …

Can you save the file to your desktop and open it with `xl_open()` just to make sure that there isn’t something else going on with your directory.

**Comment by JanMarvin** [comment]
Another idea. Could you try the chained code? In case that there’s something going wrong in the wrapper code.

**Comment by JanMarvin** [comment]
```r
wb$save(xlsx_path)
wb_load(xlsx_path)$get_mips()
```

Please check this as well, to see if the mips string is found in the file.

**Comment by PD-LUHC** [comment]
Hi @JanMarvin, thanks for you swift reply and helpful suggestions. FYI, I have a small update - because I was worried I might be doing something silly, I copied your example from here https://github.com/JanMarvin/openxlsx2/pull/983#issue-2215472480 and made two changes:

1. changed the reference file to one I generated based on our sensitivity settings
2. added a `wb_save` line pointing to the same folder that the reference file was stored in

All the intermediate validation from your example returned `TRUE`, but it still replicated the behaviour of my initial post.

Either way, thanks for your time. I'll definitely give your suggestions a whirl and report back when I can.

**Comment by JanMarvin** [comment]
Hm, you could save the file that works fine via excel and inspect if there’s something going on. (Unzip and compare the xml files line by line). If you fancy you could send me both files. Otherwise I can look into the issue next week, but no promises. I can only guess if it works and if it doesn’t, excel won’t tell me.

These mips strings are basically xml stamps like „confidential“ that only appear in environments that have (maybe this specific) mips setting enabled and Microsoft believes that this is something only businesses can have.

**Comment by JanMarvin** [comment]
Here is save (and a couple of lines more)
https://github.com/JanMarvin/openxlsx2/blob/b3f915054625c5314ee028a479443062ea1ec712/R/class-workbook.R#L2864

And wb_open uses xl_open
https://github.com/JanMarvin/openxlsx2/blob/b3f915054625c5314ee028a479443062ea1ec712/R/xl_open.R#L49

Simply saving the file and opening the workbook.

**Comment by PD-LUHC** [comment]
> Hm, you could save the file that works fine via excel and inspect if there’s something going on. (Unzip and compare the xml files line by line). If you fancy you could send me both files. Otherwise I can look into the issue next week, but no promises. I can only guess if it works and if it doesn’t, excel won’t tell me.
> 
> These mips strings are basically xml stamps like „confidential“ that only appear in environments that have (maybe this specific) mips setting enabled and Microsoft believes that this is something only businesses can have.

Yeah I think there is some corporate voodoo going on with mips. I had a look through the xml files of the two xlsx and found references to LabelInfo.xml in 2 other files (.rels and [Content_Types].xml). I found no major differences (some of the entries had a different ordering, but the entries were the same).

Interestingly I zipped the files back up and opened in excel to find that both were now showing as having no label applied. So the act of unzipping and zipping was enough to disrupt the reading/application of the label. For completeness, unzipping the re-zipped file showed exactly the same contents and `get_mips()` returns the expected value

So beyond setting the right label xml, I think maybe there is some other process/rule that is determining how LabelInfo is being read and applied.

Corporate voodoo.

**Comment by JanMarvin** [comment]
What bothers me, though. That it works for one file and not for the other.
* open: write file to tempdir and open file with spreadsheet software 
* save: write file to selected path

I just don’t see why one fails and the other works.

**Comment by JanMarvin** [comment]
You might try adding something like this (this is the only thing that I found to be different in both files, using `wb_save()` and `wb_open()` and saving from Excel), maybe Excel requires any of these default strings to be available. DocSecurity maybe?

```R
wb <- wb_workbook() |>
  wb_add_worksheet() |> 
  wb_add_mips(xml = mip_xml)

# add some default values for app.xml
wb$app$AppVersion        <- "<AppVersion>16.0300</AppVersion>"
wb$app$DocSecurity       <- "<DocSecurity>0</DocSecurity>"
wb$app$LinksUpToDate     <- "<LinksUpToDate>false</LinksUpToDate>"
wb$app$ScaleCrop         <- "<ScaleCrop>false</ScaleCrop>"
wb$app$SharedDoc         <- "<SharedDoc>false</SharedDoc>"
wb$app$HyperlinksChanged <- "<HyperlinksChanged>false</HyperlinksChanged>"

wb |> wb_save(file = xlsx_path, overwrite = TRUE) # this saves a file with no mip label set
```

Other than that I'm a little clueless to be honest. If your file contains references to `LabelInfo.xml`, there should be something in `wb$docMetadata`. If there is a file `LabelInfo.xml` in the export of one of the two files and not in the other, or if the reference is missing, the xlsx file will be broken or at least the content of docMetadata file will not be available. Sometimes it's a little hard to predict how Excel behaves and when it will complain.

There are two different ways to provide MIPS strings. As custom properties in a file `docProps/custom.xml` or as `docMetadata/LabelInfo.xml`. We try to be smart and extract either the `docMetadata` or the `docProps` variant in `wb_get_mips()`. But maybe in your case something got messed up?
Anyhow, we should not end up in a situation where a file with `wb_open()` creates docMetadata and `wb_save()` doesn't.

You could inspect the mips string in the original file further, maybe our assumptions are incorrect or we read not enough of the mips string.

Working on openxlsx2 is often a bit like searching for a needle in the haystack, but right now I do not even have a way to validate if I have found a needle.

**Comment by JanMarvin** [comment]
For further reference maybe have a look at #1028

**Comment by JanMarvin** [comment]
Last words for tonight, I guess that one workbook is the original file just loaded and the other is the modified version. Thought no idea what goes wrong. As states open is a wrapper around save. Therefore what happens in one, happens in the other.

**Comment by PD-LUHC** [comment]
Thanks for the efforts Jan - very much appreciated. Hopefully I'll get a bit of time in the week to have more of a play with this. FWIW I think the assumptions about the string are correct. Maybe the point about default values for app.xml is a good shout. Either way I'll report back if I find anything of note.

**Comment by PD-LUHC** [comment]
So I feel a little stupid - I think things may have been working all along but obscured by document "Protected View" settings. I was testing on our default configuration in which scripts output to a network drive. The network drive is not on a trusted list and so it triggers "Protected View" when opening files - it happens so often that I think my brain blanks it out. Temp files are obviously local so don't suffer from this.

It seems that mips labels don't load prior to protection policies being applied. So initially it looks like there is no security label (why MS use the same graphic for label unset vs. label not determined/loaded, is maybe another issue). Once the user acknowledges the protected view warning, Excel is then free to load broader features, MIP being one of them.

Long story short - if the user clicks "Enable Editing" then things work fine. Sorry for sending you down a rabbit hole @JanMarvin. If I can help in any other way to test any changes to mip then please let me know. Again thanks for your efforts.

**Comment by JanMarvin** [comment]
Haha, splendid! Thanks for sharing your findings. In this case it was helpful, that I am unable to test the process -> no rabbit hole, just a few comments. Glad that it works!
